### PR TITLE
PE-1136/refactor arconnect signing

### PR DESCRIPTION
**Pull request by jdaev**
Refactors arconnect wallets to use the sign function instead of getSignature function to sign any transactions and data items.

Related arweave-dart PR: https://github.com/ardriveapp/arweave-dart/pull/20

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ dependencies:
   arweave:
     git:
       url: https://github.com/ardriveapp/arweave-dart
-      ref: v3.0.1
+      ref: 3f3c72ec5e90df1076638eee7ac36310cbd371bf
```
[the remark this answers is not in the document]
Comment (karlprieb) [reply pubspec.yaml:30]: We do that on the staging step after QA. Until then, we use the checksum.

File: lib/services/arconnect/arconnect_wallet.dart [hunk 2]
```diff
@@ -17,7 +19,15 @@ class ArConnectWallet extends Wallet {
   }
 
   @override
-  Future<Uint8List> sign(Uint8List message) async{
+  Future<Uint8List> signMessage(Uint8List message) async {
```
[the remark this answers is not in the document]
Comment (jdaev) [reply lib/services/arconnect/arconnect_wallet.dart:22]: The ArConnectWallet class inherits from the arweave-dart Wallet class. I don't see a need for the rename, we are signing a message not getting a signature. getSignature implies that it's a getter on some wallet property.

File: docs/private_drive_kdf_reference.dart [hunk 3]
```diff
@@ -28,7 +27,7 @@ void main() async {
   // the same key even if the user reuses a password.
   final driveIdBytes = Uuid.parse('<drive uuid>');
   final walletSignature = await wallet
-      .sign(Uint8List.fromList(utf8.encode('drive') + driveIdBytes));
+      .signMessage(Uint8List.fromList(utf8.encode('drive') + driveIdBytes));
```
[the remark this answers is not in the document]
Comment (jdaev) [reply docs/private_drive_kdf_reference.dart:30]: See above

**Review by karlprieb — APPROVED** [review]

**Review by fedellen — DISMISSED** [review]

**Review by matibat — APPROVED** [review]
Looks good to me

**Comment by matibat** [comment]
There are conflicts with the base branch RN

**Review by thiagocarvalhodev — APPROVED** [review]

**Pull request closed without merging**